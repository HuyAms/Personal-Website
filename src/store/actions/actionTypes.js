export const FETCH_PROJECTS_START = "FETCH_PROJECTS_START";
export const FETCH_PROJECTS_SUCCESS = "FETCH_PROJECTS_SUCCESS";
export const FETCH_PROJECT_FAIL = "FETCH_PROJECT_FAIL";
export const SET_PROJECT_FILTER = "SET_PROJECT_FILTER";
export const SET_PROJECT_SERACH = "SET_PROJECT_SEARCH";

export const FETCH_INFO_START = "FETCH_INFO_START";
export const FETCH_INFO_SUCCESS = "FETCH_INFO_SUCCESS";
export const FETCH_INFO_FAIL = "FETCH_INFO_FAIL";

export const FETCH_POSITIVE_START = "FETCH_POSITIVE_START";
export const FETCH_POSITIVE_SUCCESS = "FETCH_POSITIVE_SUCCESS";
export const FETCH_POSITIVE_FAIL = "FETCH_POSITIVE_FAIL";

export const POST_POSITIVE_START = "POST_POSITIVE_START";
export const POST_POSITIVE_SUCCESS = "POST_POSITIVE_SUCCESS";
export const POST_POSITIVE_FAIL = "POST_POSITIVE_FAIL";
export const OPEN_POSITIVE_FORM = "OPEN_POSITIVE_FORM";
export const CLOSE_POSITIVE_FORM = "CLOSE_POSITIVE_FORM";