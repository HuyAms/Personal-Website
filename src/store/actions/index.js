export {
  fetchProject
} from './project';

export {
  fetchInfo
} from './info';