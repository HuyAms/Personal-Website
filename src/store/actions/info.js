import * as actionTypes from './actionTypes';

export const fetchInfoSuccess = (myInfo) => {
  return {
    type: actionTypes.FETCH_INFO_SUCCESS,
    myInfo
  }
}

export const fetchInfoFail = (error) => {
  return {
    type: actionTypes.FETCH_INFO_FAIL,
    error
  }
}

export const fetchInfoStart = () => {
  return {
    type: actionTypes.FETCH_INFO_START,
  }
}

export const fetchExpSuccess = (workExps) => {
  return {
    type: actionTypes.FETCH_EXP_SUCCESS,
    workExps
  }
}

export const fetchExpFail = (error) => {
  return {
    type: actionTypes.FETCH_EXP_FAIL,
    error
  }
}

export const fetchExpStart = () => {
  return {
    type: actionTypes.FETCH_EXP_START,
  }
}
