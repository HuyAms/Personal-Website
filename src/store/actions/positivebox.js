import * as actionTypes from './actionTypes';

export const fetchPositiveSuccess = (positives) => {
  return {
    type: actionTypes.FETCH_POSITIVE_SUCCESS,
    positives,
  };
};

export const fetchPositiveFail = (error) => {
  return {
    type: actionTypes.FETCH_POSITIVE_FAIL,
    error,
  };
};

export const fetchPositiveStart = () => {
  return {
    type: actionTypes.FETCH_POSITIVE_START,
  };
};

export const postPositiveSuccess = (positiveData, id) => {
  return {
    type: actionTypes.POST_POSITIVE_SUCCESS,
    positiveData,
    id,
  };
};

export const postPositiveFail = (error) => {
  return {
    type: actionTypes.POST_POSITIVE_FAIL,
    error,
  };
};

export const postPositiveStart = (positiveData) => {
  return {
    type: actionTypes.POST_POSITIVE_START,
    positiveData
  };
};

export const openPositiveForm = () => {
  return {
    type: actionTypes.OPEN_POSITIVE_FORM
  }
};

export const closePositiveForm = () => {
  return {
    type: actionTypes.CLOSE_POSITIVE_FORM
  }
};
