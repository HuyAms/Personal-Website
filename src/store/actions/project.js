import * as actionTypes from './actionTypes';
import axios from '../../axious';

export const fetchProjectsSuccess = (projects) => {
  return {
    type: actionTypes.FETCH_PROJECTS_SUCCESS,
    projects: projects
  }
}

export const fetchProjectsFail = (error) => {
  return {
    type: actionTypes.FETCH_PROJECT_FAIL,
    error: error
  }
}

export const fetchProjectsStart = () => {
  return {
    type: actionTypes.FETCH_PROJECTS_START
  }
}

export const setProjectFilter = (filter) => {
  return {
    type: actionTypes.SET_PROJECT_FILTER,
    filter: filter
  }
}

export const fetchProject = () => {
  return dispatch => {
    dispatch(fetchProjectsStart());
    axios.get('/projects.json')
    .then(res => {
      const fetchedProject = [];
      for (let key in res.data) {
        fetchedProject.push({
          ...res.data[key],
          id: key,
        });
      }
      dispatch(fetchProjectsSuccess(fetchedProject))
    })
    .catch(error => {
        dispatch(fetchProjectsFail(error))
    })
  }
}
