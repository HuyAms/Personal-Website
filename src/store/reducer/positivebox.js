import * as actionTypes from '../actions/actionTypes';
import updateObject from '../utility';

const initialState = {
  positives: [],
  loading: false,
  error: null,
  sharing: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.FETCH_POSITIVE_START:
      return fetchPositiveStart(state, action);
    case actionTypes.FETCH_POSITIVE_FAIL:
      return fetchPositiveFail(state, action);
    case actionTypes.FETCH_POSITIVE_SUCCESS:
      return fetchPositiveSuccess(state, action);
    case actionTypes.POST_POSITIVE_START:
      return postPositiveStart(state, action);
    case actionTypes.POST_POSITIVE_FAIL:
      return postPositiveFail(state, action);
    case actionTypes.POST_POSITIVE_SUCCESS:
      return postPositiveSuccess(state, action);
    case actionTypes.OPEN_POSITIVE_FORM:
      return openPositiveForm(state, action);
    case actionTypes.CLOSE_POSITIVE_FORM:
      return unSharingPositive(state, action);
    default:
      return state;
  }
};

const openPositiveForm = (state, action) => {
  return updateObject(state, {
    sharing: true,
  });
};

const unSharingPositive = (state, action) => {
  return updateObject(state, {
    sharing: false,
  });
};

const postPositiveStart = (state, action) => {
  return updateObject(state, {
    error: null,
    loading: true
  });
};

const postPositiveFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
  });
};

const postPositiveSuccess = (state, action) => {
  const newPositive = {
    ...action.positiveData,
    id: action.id,
  };

  return updateObject(state, {
    positives: state.positives.concat(newPositive),
    loading: false,
    sharing: false
  });
};

const fetchPositiveStart = (state, action) => {
  return updateObject(state, {
    error: null,
    loading: true,
  });
};

const fetchPositiveFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
  });
};

const fetchPositiveSuccess = (state, action) => {
  return updateObject(state, {
    positives: action.positives,
    loading: false,
  });
};

export default reducer;