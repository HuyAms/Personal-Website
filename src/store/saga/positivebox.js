import {put, call} from 'redux-saga/effects';
import * as actions from '../actions/positivebox';
import axios from "../../axious";

export function* fetchPositiveSaga(action) {
  try {

    const response = yield call(axios.get, '/positive.json');

    const fetchedPositive = [];
    for (let key in response.data) {
      fetchedPositive.push({
        ...response.data[key],
        id: key,
      });
    }

    yield put(actions.fetchPositiveSuccess(fetchedPositive));

  } catch(error) {

    yield put(actions.fetchPositiveFail(error));
  }
}

export function* postPositivesSaga(action) {
  try {

    const response = yield call(axios.post, '/positive.json', action.positiveData);

    yield put(actions.postPositiveSuccess(action.positiveData, response.data.name));

  } catch(error) {

    yield put(actions.postPositiveFail(error));
  }
}
