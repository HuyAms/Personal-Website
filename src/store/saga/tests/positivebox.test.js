/**
 * Tests for PositiveBox sagas
 */

import {put} from 'redux-saga/effects';
import {fetchPositiveSaga, postPositivesSaga} from '../positivebox';
import * as actions from '../../actions/positivebox';

describe('fetchPositiveSaga', () => {
  let fetchPositiveGenerator;

  beforeEach(() => {
    fetchPositiveGenerator = fetchPositiveSaga();

    const callDescriptor = fetchPositiveGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the fetchPositiveSuccess action if it requests the data successfully', () => {
    const response = {
      data: []
    }

    const putDescriptor = fetchPositiveGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(actions.fetchPositiveSuccess(response.data)));
  })

  it('should dispatch the fetchPositiveFail action if the response errors', () => {
    const error = new Error('Some error');
    const putDescriptor = fetchPositiveGenerator.throw(error).value;
    expect(putDescriptor).toEqual(put(actions.fetchPositiveFail(error)));
  })
})

describe('postPositivesSaga', () => {
  let postPositiveGenerator;

  let action = {
    positiveData: []
  }

  beforeEach(() => {
    postPositiveGenerator = postPositivesSaga(action);

    const callDescriptor = postPositiveGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the postPositiveSuccess action if it requests the data successfully', () => {
    const response = {
      data: {
        name: ""
      }
    }

    const putDescriptor = postPositiveGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(actions.postPositiveSuccess(action.positiveData, response.data.name)));
  })

  it('should dispatch the postPositiveFail action if the response errors', () => {
    const error = new Error('Some error');
    const putDescriptor = postPositiveGenerator.throw(error).value;
    expect(putDescriptor).toEqual(put(actions.postPositiveFail(error)));
  })
})