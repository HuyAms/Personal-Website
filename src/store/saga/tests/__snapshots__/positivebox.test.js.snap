// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPositiveSaga should dispatch the fetchPositiveFail action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/positive.json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetchPositiveSaga should dispatch the fetchPositiveSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/positive.json",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`postPositivesSaga should dispatch the postPositiveFail action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/positive.json",
      Array [],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`postPositivesSaga should dispatch the postPositiveSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/positive.json",
      Array [],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
